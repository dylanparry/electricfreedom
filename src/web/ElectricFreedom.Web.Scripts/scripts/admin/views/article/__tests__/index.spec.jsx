import React from 'react';
import { object } from 'prop-types';
import createContext from 'react-router-test-context';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import ArticleSwitch from '../index';
import ArticleAdd from '../add';
import ArticleEdit from '../edit';
import ArticleList from '../list';

const setup = (pathname = '/') =>
{
  const options = {
    context: createContext({
      location: { pathname },
    }),
    childContextTypes: {
      router: object,
    },
  };
  const wrapper = mount(<ArticleSwitch />, options);

  return {
    wrapper,
    snapshot: toJson(wrapper),
  };
};

describe('ArticleSwitch', () =>
{
  test('render', () =>
  {
    const { snapshot } = setup();

    expect(snapshot).toMatchSnapshot();
  });

  test('route "/articles/add" renders ArticleAdd', () =>
  {
    const { wrapper } = setup('/articles/add');

    expect(wrapper.find(ArticleAdd).exists()).toBe(true);
  });

  test('route "/articles/:articleId" renders ArticleEdit', () =>
  {
    for (let i = 1; i <= 5; i += 1)
    {
      const { wrapper } = setup(`/articles/${i}`);

      const articleEdit = wrapper.find(ArticleEdit);
      expect(articleEdit.exists()).toBe(true);
      expect(articleEdit.props().match.params.articleId).toBe(i.toString());
    }
  });

  test('route "/articles" renders ArticleList', () =>
  {
    const { wrapper } = setup('/articles');

    expect(wrapper.find(ArticleList).exists()).toBe(true);
  });
});
