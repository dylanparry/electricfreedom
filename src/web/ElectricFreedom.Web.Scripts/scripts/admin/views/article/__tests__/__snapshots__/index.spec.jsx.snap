// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleSwitch render 1`] = `
<articleSwitch>
  <Switch />
</articleSwitch>
`;
